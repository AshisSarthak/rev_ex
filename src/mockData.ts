export const demoPacketData = {
  balance: 100,
  transactions: [
    {
      id: 1,
      timestamp: 1000230,
      balance: 50,
    },
    {
      id: 2,
      timestamp: 1000230,
      balance: 80,
    },
    {
      id: 3,
      timestamp: 1000230,
      balance: 90,
    },
  ],
};

export enum CURRENCY_SYMBOL {
  USD = "$",
  EUR = "€",
  GBP = "£",
}

export enum CURRENCYS {
  USD = "USD",
  EUR = "EUR",
  GBP = "GBP",
}
