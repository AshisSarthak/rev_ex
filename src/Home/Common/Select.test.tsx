import React from "react";
import { shallow } from "enzyme";
import SelectEl from "./Select";
import { CURRENCYS } from "../../mockData";

describe("Select Element", () => {
  const mockFn = jest.fn();

  it("Loads the default Dropdown", () => {
    const wrapper = shallow(
      <SelectEl
        currentVal={CURRENCYS.EUR}
        optionLists={[CURRENCYS.EUR, CURRENCYS.GBP, CURRENCYS.USD]}
        onChange={mockFn}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Loads the Dropdown", () => {
    const wrapper = shallow(
      <SelectEl
        currentVal={CURRENCYS.EUR}
        optionLists={[CURRENCYS.EUR, CURRENCYS.GBP, CURRENCYS.USD]}
        onChange={mockFn}
      />
    );
    const select = wrapper.find("select").at(0);
    expect(select.prop("value")).toBe(CURRENCYS.EUR);
    select.simulate("change", { target: { value: CURRENCYS.GBP } });
    expect(mockFn).toBeCalledWith(CURRENCYS.GBP);
  });
});
