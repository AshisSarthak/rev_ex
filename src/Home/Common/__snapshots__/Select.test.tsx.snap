// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Element Loads the default Dropdown 1`] = `
<select
  className="exchange-from-select"
  onChange={[Function]}
  value="EUR"
>
  <option
    key="EUR"
    value="EUR"
  >
    EUR
  </option>
  <option
    key="GBP"
    value="GBP"
  >
    GBP
  </option>
  <option
    key="USD"
    value="USD"
  >
    USD
  </option>
</select>
`;
